src/frontend/src/lib/validation/utils.validation.ts

```diff
@@ -0,0 +1,19 @@
+import { ZodType, type ZodTypeDef } from 'zod';
+
+interface SafeParseParams<T, Fallback> {
+	schema: ZodType<T, ZodTypeDef, unknown>;
+	value: unknown;
+	fallback?: Fallback;
+}
+
+type SafeParseReturn<T, Fallback> = Fallback extends undefined ? T | undefined : T;
+
+export const safeParse = <T, Fallback extends T | undefined = undefined>({
+	schema,
+	value,
+	fallback = undefined
+}: SafeParseParams<T, Fallback>): SafeParseReturn<T, Fallback> => {
+	const { success, data } = schema.safeParse(value);
+
+	return success ? data : (fallback as SafeParseReturn<T, Fallback>);
```

Comment:
I guess no other way to avoid casting?

Reply:
unfortunately not... I tried/searched a lot and it seems a limitation of the conditional typing; however it works: simulating the exact return type for the cases where we specify a `fallback`

![Screenshot 2024-11-01 at 22 05 08](https://github.com/user-attachments/assets/94450b5f-b539-4e32-9c11-eb90ef23947b)
![Screenshot 2024-11-01 at 22 05 18](https://github.com/user-attachments/assets/2943be32-3794-4b81-ba95-c7d953b0d28b)

and if we remove the fallback